airbyte-integrations/connectors/destination-chroma/destination_chroma/indexer.py

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from typing import List
+
+import chromadb
+from chromadb.config import Settings
+from airbyte_cdk.destinations.vector_db_based.document_processor import METADATA_RECORD_ID_FIELD, METADATA_STREAM_FIELD, Chunk
+from airbyte_cdk.destinations.vector_db_based.embedder import Embedder
+from airbyte_cdk.destinations.vector_db_based.indexer import Indexer
+from airbyte_cdk.destinations.vector_db_based.utils import format_exception
+from airbyte_cdk.models import ConfiguredAirbyteCatalog
+from airbyte_cdk.models.airbyte_protocol import DestinationSyncMode
+
+from destination_chroma.config import ChromaIndexingConfigModel
+from .utils import validate_collection_name
+
+
+class ChromaIndexer(Indexer):
+    def __init__(self, config: ChromaIndexingConfigModel, embedder: Embedder):
+        super().__init__(config, embedder)
+        self.client = self._get_client()
+        self.collection_name = validate_collection_name(config.collection_name)
+
+    def check(self):
+        try:
+            heartbeat = self.client.heartbeat()
+            if not heartbeat:
+                return "Chroma client server is not alive"
+            collection = self.client.get_or_create_collection(self.collection_name)
+            count = collection.count()
+            if count!=0 and not count:
+                return f"unable to get or create collection with name {self.collection_name}"
+            return
+        except Exception as e:
+            return format_exception(e)
+
+    def index(self, document_chunks: List[Chunk], delete_ids: List[str]) -> None:
+        if len(delete_ids) > 0:
+            self._delete_records(delete_ids=delete_ids)
+        embedding_vectors = self.embedder.embed_texts([chunk.page_content for chunk in document_chunks])
+        entities = []
+        for i in range(len(document_chunks)):
+            chunk = document_chunks[i]
+            entities.append({'id': chunk.metadata[METADATA_RECORD_ID_FIELD], 
```

Comment:
As there might be multiple chunks per record id, you can't use them as the id for the Chroma entity. Instead, generate a uuid. When deleting entities via the record id, use the delete by filter approach like for the stream instead of deleting by id (just on the METADATA_RECORD_ID_FIELD instead of METADATA_STREAM_FIELD)